### PR TITLE
Suppress most warnings when compiling with clang.

### PR DESCRIPTION
**Pull request by MarcelPiNacy**
For builds with clang aiming to use -Wall or -Werror, this commit fixes or suppresses the following warnings:
- "-Wcast-align" is suppressed through pragma.
- "-Wdeclaration-after-statement" is suppressed through pragma.
- "-Wreserved-identifier" for functions that start with '`_`' are prefixed now with '`svb_`'.
- "-Wsign-conversion" through explicit casting, changing integer literals or, as is the case of the lookup tables, by changing type directly.
- "-Wstrict-prototypes" through adding void ("f()" -> "f(void)").
- "-Wcast-qual" through explicit casting.
Note: ARM code has not been modified yet.

### CODE REVIEW

**Comment by lemire** [comment]
As part of this PR, are you game to add something like this to the CMakeLists.txt?


```CMake
  if(NOT (MSVC)
         add_compile_options("-Wall")
  endif()
```

Or some appropriate setting? Then when we are good, we can pass -Werror to CI. This would prevent future regressions.

E.g.,

```
CXXFLAGS=-Werror cmake -B build
```

**Comment by lemire** [comment]
There are builds errors.

**Comment by MarcelPiNacy** [comment]
It seems I forgot to check for C11 atomics support, now it should be fixed.
I have also added STREAMVBYTE_WALL and STREAMVBYTE_WERROR to pass the flags from CMake.

**Comment by MarcelPiNacy** [comment]
There is still an issue with the CMakeLists.txt. Sorry, give me a moment.

**Comment by MarcelPiNacy** [comment]
The flags weren't being actually passed to the compiler. It's fixed now.

**Comment by MarcelPiNacy** [comment]
Testing Wall and Werror in CI.

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,40 @@ set(STREAMVBYTE_LIB_SOVERSION "1" CACHE STRING "streamvbyte library soversion")
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 
+include(CheckCCompilerFlag)
+
+option(STREAMVBYTE_WALL "Emmit all warnings during compilation." ON)
```
Comment (lemire) [inline CMakeLists.txt:14]: Emmit -> emit

**Comment by lemire** [comment]
Unsurprisingly, there are some warnings.

**Comment by MarcelPiNacy** [comment]
It looks like for MSVC builds spectre-related warnings are inserted that behave more like annotations. They are being suppressed in CMakeLists.txt.
There are many warnings emitted in the CI that are from the corecrt. I'm not sure what to do here.

**Comment by lemire** [comment]
You have set `/Wall` which is like turning the dial to 15.

I have been using something like this...

```CMake

  if("${MSVC_TOOLSET_VERSION}" STRLESS_EQUAL "140")
    target_compile_options(something something /W0 /sdl)
  else()
    target_compile_options(something something /WX /W3 /sdl /w34714)
  endif()
```

**Comment by MarcelPiNacy** [comment]
I've switched to the flags you mentioned for MSVC builds.

**Comment by lemire** [comment]
Running tests

**Comment by lemire** [comment]
@MarcelPiNacy Consider replacing `-2147483648` by `-2147483647 - 1` in unit to silence the Visual Studio warning.

Note that the code is correct. The warning, like many Visual Studio warning, is misleading.

**Comment by MarcelPiNacy** [comment]
Let's see if this fixes the VS warning...

**Comment by MarcelPiNacy** [comment]
This should suppress the remaining warnings related to unused variables.

**Comment by lemire** [comment]
Running tests.

**Comment by lemire** [comment]
Yé!

@MarcelPiNacy Do you recommend merging ?!

**Comment by MarcelPiNacy** [comment]
That should be fine now. I'll open another PR if something else comes up.

**Comment by lemire** [comment]
Merging.

**Pull request merged**